### PR TITLE
Fix NCCL deadlock in Ulysses SP when sequence length has remainder

### PR DESCRIPTION
**Pull request by storyicon**
## Motivation
The issue is a deadlock that occurs in Ulysses sequence-parallel (SP) mode when the audio/video sequence length is not evenly divisible by `sp_world_size`.

`_build_ltx2_sp_padding_mask` returns `None` when a rank has no padding tokens (`valid >= seq_len`), but returns a mask `Tensor` for ranks that do have padding. Since `USPAttention.forward` branches on `attn_mask is not None`, different ranks execute different sets of NCCL collective operations:

- **No-padding ranks** (mask=None): skip `all_gather(mask)`, go directly to local attention then `all-to-all output`
- **Padding rank** (mask=Tensor): call `all_gather(mask)` first, then attend, then `all-to-all output`

The padding rank blocks on `all_gather(mask)` waiting for the other ranks to join, while those ranks never call it. This causes a deadlock.

## Modifications
Remove the `valid >= seq_len → None` early return in `_build_ltx2_sp_padding_mask`. When there is no padding, return an all-ones mask Tensor instead of `None`, so all ranks take the same masked-attention path through `USPAttention`.

The all-ones mask is semantically equivalent to `None` (no tokens are masked out), so this change has no effect on model output. It only ensures NCCL collective operations are consistent across ranks.

## Accuracy Tests
No accuracy impact. The all-ones mask is semantically equivalent to `None`, no tokens are masked out.

### CODE REVIEW

**Comment by storyicon** [comment]
I've been working on integrating the LTX2.3 pipeline into SGLang. During testing, I kept running into NCCL timeouts, which puzzled me for a while. I initially suspected there was something wrong with inter-GPU communication on my machine. While investigating again, finally found the root cause. cc: @mickqian

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/ltx_2_denoising.py [hunk 1]
```diff
@@ -1050,12 +1050,13 @@ def _build_ltx2_sp_padding_mask(
...
-        if valid <= 0 or valid >= int(seq_len):
+        if valid <= 0:
             return None
         mask = torch.ones(
             (batch_size, int(seq_len)), device=device, dtype=torch.float32
         )
-        mask[:, valid:] = 0.0
+        if valid < int(seq_len):
+            mask[:, valid:] = 0.0
         return mask
```
[the remark this answers is not in the document]
Comment (storyicon) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/ltx_2_denoising.py]: Good catch on both. Fixed: 
1. also removed the valid <= 0 returning None branch since some SP ranks can have zero valid tokens; 
2. switched to torch.bool since float32 1.0/0.0 is incorrect as SDPA additive mask, _prepare_sdpa_mask converts bool True to 0.0 and False to neg_inf correctly.

**Review by mickqian — APPROVED** [review]

**Comment by mickqian** [comment]
@storyicon thanks so much! that said, we prefer cfg-parallel + tp for LTX model deployment, see: https://docs.sglang.io/cookbook/diffusion/LTX/LTX2%20&%20LTX2.3#3-1-basic-configuration

**Comment by storyicon** [comment]
> @storyicon thanks so much! that said, we prefer cfg-parallel + tp for LTX model deployment, see: https://docs.sglang.io/cookbook/diffusion/LTX/LTX2%20&%20LTX2.3#3-1-basic-configuration

Appreciate the fast response! I'll try out your suggestion. Thanks for all your work on sgl-diffusion and for the LTX support.
